src/core/views/popup.js

```diff
@@ -56,25 +60,18 @@ const Overlay = styled.div`
   z-index: 10001;
   min-height: 100%;
 `
+const dimensions = css`
+  ${({ size }) => size && `
+    height: calc(${size}% - 80px); 
+    width: calc(${size}% - 80px); 
+  `}
+`
 
 const popupWrapper = styled(BaseWrapper)`
-  ${p => {
-    const offset = (100 - p.size) / 2
-    if (p.isContained) {
-      return `
-        width: calc(${p.size}% - 80px);
-        height: calc(${p.size}% - 80px);
-        top: calc(${offset}% + 40px);
-        left: calc(${offset}% + 40px);
-      `
-    }
-    return `
-      width: calc(${p.size}% - 80px);
-      height: calc(${p.size}% - 80px);
-      top: calc(${offset}% + 40px);
-      left: calc(${offset}% + 40px);
-    `
-  }}
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);\${({})}
+  ${({ size }) => size && dimensions}  
```

Comment:
Great job removing that complicated calculations!

One minor thing - I would not keep the `dimensions` in separate variable since its not reused anywhere.